provider/pkg/pulumiapi/schedules.go

```diff
@@ -104,6 +105,9 @@ func (c *Client) GetSchedule(ctx context.Context, stack StackIdentifier, schedul
 	var scheduleResponse ScheduleResponse
 	_, err := c.do(ctx, http.MethodGet, apiPath, nil, &scheduleResponse)
 	if err != nil {
+		if strings.Contains(err.Error(), "404 API error") {
```

Comment:
I think you should check the http error code rather than doing a string match on the error message. c.do will [return an error](https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/pulumiapi/client.go#L94) that carries along the http status code, and there is a [helper to check it](https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/pulumiapi/error.go#L31):

```suggestion
		if GetErrorStatusCode(err) == http. {
```

Reply:
Ah, good idea, completely forgot about this method! Will update to use `GetErrorStatusCode(err) == http.StatusNotFound`